storage/innobase/row/row0merge.cc

```diff
@@ -2258,7 +2271,11 @@ row_merge_read_clustered_index(
 				goto func_exit;
 			}
 
-			ulonglong	value = sequence++;
+			ulonglong value;
+			if (likely(!historical_row))
+				value = sequence++;
+                        else
+				value = historic_auto_decrement--;
```

Comment:
How does the `historic_auto_decrement` work with `auto_increment_increment` or `auto_increment_offset`?

Please add a source code comment to describe what the desired behaviour is when adding an `AUTO_INCREMENT` column to a system versioned table. To me, it would seem to be desired to assign the same `AUTO_INCREMENT` value for each historical value of a record, instead of assigning values from a decrementing sequence that looks like an ad-hoc solution.

Reply:
It has no connection with `auto_increment_increment` and `auto_increment_offset` as it solves different problem: unique constraint. The value in historical rows doesn't matter in this case as it was not existed at that time. Probably `NULL` value would do a better job.